### PR TITLE
[1LP][RFR] SSA host update

### PR DESCRIPTION
**Pull request by rlbabyuk**
Purpose or Intent
=================
Update SSA host test

{{pytest: -k test_run_host_analysis -vvv --long-running --use-provider complete}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 091c77548932364e8eca4c46211d59da693a2b73:

`cfme/tests/infrastructure/test_host_analysis.py`:
- :bangbang: Line 14:5: **E265** *block comment should start with '# '*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d753860fc7cce0b1c79b7736c44b982ad2823528:

`cfme/tests/infrastructure/test_host_analysis.py`:
- :red_circle: Line 7:1: **F401** *'cfme.infrastructure.provider.virtualcenter.VMwareProvider' imported but unused*
- :red_circle: Line 34:13: **F631** *assertion is always true, perhaps remove parentheses?*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b9c3d196ec8fb9b0538ca5b737b53f863b68bff8:

`cfme/tests/infrastructure/test_host_analysis.py`:
- :red_circle: Line 7:1: **F401** *'cfme.infrastructure.provider.virtualcenter.VMwareProvider' imported but unused*
- :red_circle: Line 34:13: **F631** *assertion is always true, perhaps remove parentheses?*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/infrastructure/test_host_analysis.py [hunk 1]
```diff
@@ -19,110 +14,78 @@
...
 
 
-def get_host_data_by_name(provider_key, host_name):
-    for host_obj in conf.cfme_data.get('management_systems', {})[provider_key].get('hosts', []):
-        if host_name == host_obj['name']:
-            return host_obj
-    return None
-
+@pytest.fixture(scope='module')
+def host_with_credentials(appliance, provider, host_name):
```
Comment (quarckster) [inline cfme/tests/infrastructure/test_host_analysis.py]: **REQUIRED**
You can use this function https://github.com/ManageIQ/integration_tests/blob/master/cfme/utils/hosts.py#L21

File: cfme/infrastructure/host.py [hunk 2]
```diff
@@ -269,7 +271,7 @@ def get_db_id(self):
         else:
             return self.db_id
 
-    def run_smartstate_analysis(self):
+    def run_smartstate_analysis(self, check_task_result=False):
```
Comment (quarckster) [inline cfme/infrastructure/host.py]: **REQUIRED**
I would replace `check_task_result` by `wait_for_task_result`

File: cfme/tests/infrastructure/test_host_analysis.py (hunk 1 above)
Comment (rlbabyuk) [reply cfme/tests/infrastructure/test_host_analysis.py]: Yes, it is an option, but I would rather update cheds via API here

File: cfme/tests/infrastructure/test_host_analysis.py [hunk 3]
```diff
@@ -19,110 +14,78 @@
...
+        soft_assert(host_with_credentials.get_detail('Configuration', 'Packages') != '0',
+                    'No packages found in host detail')
+        soft_assert(host_with_credentials.get_detail('Configuration', 'Files') != '0',
+                    'No files found in host detail')
 
     elif host_type in ('esx', 'esxi'):
-        soft_assert(test_host.get_detail('Configuration', 'Advanced Settings') != '0',
-            'No advanced settings found in host detail')
-
-        if not(provider.type == "virtualcenter" and provider.version < "5"):
```
Comment (mshriver) [inline cfme/tests/infrastructure/test_host_analysis.py]: :+1:

**Comment by mshriver** [comment]
Looks good, but some softassert failures in 5.9 for the host firewall rules (run 19984)

File: cfme/tests/infrastructure/test_host_analysis.py (hunk 1 above)
Comment (mshriver) [reply cfme/tests/infrastructure/test_host_analysis.py]: I'm okay with this since the utils method doesn't use rest.

**Pull request merged**